presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/CommandExecutor.java

```diff
@@ -13,25 +13,55 @@
  */
 package com.teradata.prestomanager.agent;
 
+import org.apache.commons.io.IOUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.StringTokenizer;
 import java.util.concurrent.TimeUnit;
 
-import static java.lang.Runtime.getRuntime;
 import static java.lang.String.format;
 
-// TODO: Add logger
 public final class CommandExecutor
 {
+    private static final Logger LOGGER = LogManager.getLogger(CommandExecutor.class);
+
     private CommandExecutor() {}
 
     public static int executeCommand(String command)
             throws PrestoManagerException
     {
+        LOGGER.debug("Command to be executed: {}", command);
+
+        StringTokenizer tokenizer = new StringTokenizer(command);
+        String[] commandArray = new String[tokenizer.countTokens()];
+        for (int i = 0; tokenizer.hasMoreTokens(); i++) {
+            commandArray[i] = tokenizer.nextToken();
+        }
+
+        ProcessBuilder processBuilder = new ProcessBuilder(commandArray);
+        processBuilder.redirectErrorStream(true);
         try {
-            Process process = getRuntime().exec(command);
-            process.waitFor(2, TimeUnit.MINUTES);
+            Process process = processBuilder.start();
+            try (InputStream processStream = process.getInputStream()) {
+                new Thread(() -> {
+                    try {
+                        LOGGER.info("Output from command : {} \n {}", command, new String(IOUtils.toByteArray(processStream)));
+                    }
+                    catch (IOException e) {
+                        LOGGER.error("Failed to log the process output");
+                    }
+                }).start();
+                if (!process.waitFor(2, TimeUnit.MINUTES)) {
+                    process.destroyForcibly();
```

Comment:
I would try using `destroy()` first. Then wait for another short time, check if it's alive, and destroy it forcibly if it is (and log that it resisted the first attempt).

With the builder idea, the delay between the two calls should be settable.